### PR TITLE
Add return type to testing

### PR DESCRIPTION
**Pull request by mosi-kha**
# Note
It's my first PR 
# Summary of Changes
Declare return type [testing's functions](https://falcon.readthedocs.io/en/stable/api/testing.html)
Update .gitignore (add pycharm env)
# Related Issues

[falcon#1350](https://github.com/falconry/falcon/issues/1350)
# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://github.com/falconry/falcon/blob/master/CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] Added **tests** for changed code.
- [ ] Prefixed code comments with GitHub nick and an appropriate prefix.
- [ ] Coding style is consistent with the rest of the framework.
- [ ] Updated **documentation** for changed code.
    - [ ] Added docstrings for any new classes, functions, or modules.
    - [ ] Updated docstrings for any modifications to existing code.
    - [ ] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [ ] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://pypi.org/project/towncrier/) news fragments under `docs/_newsfragments/`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: falcon/testing/helpers.py [hunk 1]
```diff
@@ -86,8 +84,7 @@ def rand_string(min, max):
 def create_environ(path='/', query_string='', protocol='HTTP/1.1',
                    scheme='http', host=DEFAULT_HOST, port=None,
                    headers=None, app='', body='', method='GET',
-                   wsgierrors=None, file_wrapper=None, remote_addr=None):
-
+                   wsgierrors=None, file_wrapper=None, remote_addr=None) -> dict:
```
Comment (nZac) [inline falcon/testing/helpers.py]: This should return a `Dict[str, Any]`

File: falcon/testing/helpers.py [hunk 2]
```diff
@@ -249,10 +246,11 @@ def closed_wsgi_iterable(iterable):
     Returns:
         iterator: An iterator yielding the same bytestrings as `iterable`
     """
+
     def wrapper():
         try:
             for item in iterable:
-                yield item
+                    yield item
```
Comment (nZac) [inline falcon/testing/helpers.py]: I don't think the indentation here is necessary.

File: falcon/testing/client.py [hunk 3]
```diff
@@ -103,31 +101,31 @@ def __init__(self, iterable, status, headers):
...
         return self._status
 
     @property
-    def status_code(self):
+    def status_code(self) -> int:
         return self._status_code
 
     @property
-    def headers(self):
+    def headers(self) -> CaseInsensitiveDict:
```
[the remark this answers is not in the document]
Comment (nZac) [reply falcon/testing/client.py]: I think I might be misguided here, [reading the docs for mypy](https://mypy.readthedocs.io/en/latest/generics.html#defining-sub-classes-of-generic-classes)

I think we would have to define `CaseInsensitiveDict` to be generic over the value but not the key. Something like this:

```python
from typing import Mapping, TypeVar

T = TypeVar('T')

class CaseInsensitiveDict(Mapping[str, T]):
    # ....
```

That should allow us to do `CaseInsensitiveDict[str]` in this case and `CaseInsensitiveDict[list]` if we ever need something like that.

**Comment by kgriffs** [comment]
Hi @mosi-kha, have you had a chance to look at @nZac's comments? If you have the bandwidth to work on this over the next week or so, that would be awesome. Otherwise, just let me know and I would be happy to contribute additional patches to get this over the finish line.

**Comment by mosi-kha** [comment]
Hi @kgriffs @nZac 
I apologize for the delay in responding.
I like and want to contributing in the falcon because i love it❤️.
if have more time for doing this I’ll be happy.
And it’s will be my first contribution in big project.😃

**Comment by kgriffs** [comment]
Thanks for contributing! We are glad to have your help.

**Comment by kgriffs** [comment]
@nZac I've attempted to address your comments

**Pull request merged**